applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nonlinear_strain_energy_response_function.cpp

```diff
@@ -0,0 +1,385 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Mahmoud Sesa, https://github.com/mahmoudsesa
+//
+
+// System includes
+
+// External includes
+#include "utilities/variable_utils.h"
+
+// Project includes
+#include "adjoint_nonlinear_strain_energy_response_function.h"
+
+namespace Kratos
+{
+    AdjointNonlinearStrainEnergyResponseFunction::AdjointNonlinearStrainEnergyResponseFunction(ModelPart& rModelPart, Parameters ResponseSettings)
+    : AdjointStructuralResponseFunction(rModelPart, ResponseSettings)
+    {
+        for(auto cond_it = rModelPart.ConditionsBegin(); cond_it != rModelPart.ConditionsEnd(); ++cond_it)
+        {
+            mConditions[cond_it->Id()] = cond_it;
+            SizeType number_of_nodes = cond_it->GetGeometry().size();
+            SizeType dimension = cond_it->GetGeometry().WorkingSpaceDimension();
+            SizeType vec_size = number_of_nodes * dimension;
```

Comment:
Right now the size of the RHS is like you compute it here. In the future this could be changing because we recently discussed about reworking some of the conditions in a way that they are energetic consistent. This will change the size of the RHS of some conditions.

Reply:
I will add it to my ToDo list